client/js/widgets/widget.js

```diff
@@ -441,15 +442,15 @@ export class Widget extends StateManaged {
     return [ 'borderRadius', 'css', 'height', 'inheritChildZ', 'layer', 'width' ].concat(this.propertiesUsedInCSS);
   }
 
-  cssReplaceProperties(css) {
+  cssReplaceProperties(css, usedProperties) {
     for(const match of String(css).matchAll(/\$\{PROPERTY ([A-Za-z0-9_-]+)\}/g)) {
       css = css.replace(match[0], this.get(match[1]));
-      this.propertiesUsedInCSS.push(match[1]);
+      usedProperties.add(match[1]);
     }
     return css;
   }
 
-  cssToStylesheet(css, nested = false) {
+  cssToStylesheet(css, usedProperties, nested = false) {
```

Comment:
@ArnoldSmith86 I was torn between adding a parameter to track used properties or changing the return value to be an object. I think the latter may be slightly cleaner from an API perspective but this is technically slightly more efficient (don't have to create and merge sets) and a bit of a smaller code change. Wanted to get your thoughts. WDYT?

I.e. the two options are:
1. As is in this PR, usedProperties is a Set pased in that the css methods add to when a property is referenced.
2. Or the css methods return a structure such as { css: 'original css text return value', used: <set of used properties> }